### PR TITLE
release/1.8.0: document environment naming conventions

### LINKED ISSUE #1291: Standard naming convention for each environment
**Is your feature request related to a problem? Please describe.**
Each environment (e.g. unified-dev, gsi-addon, etc) should have a standard naming convention.

**Describe the solution you'd like**
Just a suggestion, I'm open to any other solutions.  I suggest replacing the template `dev` with `env`, i.e.:
unified-dev -> unified-env
gsi-addon-dev -> gsi-addon-env

If new installations are required in between official releases, append the same descriptor across all installations.  For instance, if a new version of ESMF is required:
unified-dev -> unified-env-esmfxxx
gsi-addon-dev -> gsi-addon-env-esmfxxx

**Additional context**
This would be helpful when upgrading spack-stack versions within codes used on multiple systems such that we would only need to specify a root location for spack-stack and a version number.

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

Resolves #1291 by adding a table to `doc/source/PreConfiguredSites.rst`.

### Testing

Readthedocs build: https://spack-stack--1300.org.readthedocs.build/en/1300/PreConfiguredSites.html

### Applications affected

None

### Systems affected

None

### Dependencies

None

### Issue(s) addressed

Resolves #1291 

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [ ] ~~These changes have been tested on the affected systems and applications.~~
- [ ] ~~All dependency PRs/issues have been resolved and this PR can be merged.~~

### CODE REVIEW

**Comment by srherbener** [comment]
@climbfuji is this PR supposed to be based on `release/1.8.0` instead of `develop`?

**Comment by climbfuji** [comment]
> @climbfuji is this PR supposed to be based on `release/1.8.0` instead of `develop`?

Yes :-) and now it is. Thanks for catching this!

File: configs/sites/tier1/jet/mirrors.yaml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+      url: file:///contrib/spack-stack/mirror
       access_pair:
       - null
       - null
       access_token: null
       profile: null
       endpoint_url: null
     push:
-      url: file:///mnt/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/source-cache
+      url: file:///contrib/spack-stack/mirror
```
Comment (AlexanderRichert-NOAA) [inline configs/sites/tier1/jet/mirrors.yaml]: My understanding is that space on /contrib is a bit more precious than lfs. Not that we're not allowed to put caches there, but from a being-good-neighbors perspective we might want to consider leaving the caches on lfs, and just have the installations themselves under /contrib.

**Comment by AlexanderRichert-NOAA** [comment]
Documentation updates looks good, thanks for adding that!

**Comment by climbfuji** [comment]
The jet updates have nothing to do with this PR and should be removed ...

**Comment by RatkoVasic-NOAA** [comment]
> The jet updates have nothing to do with this PR and should be removed ...
@climbfuji , sorry, I saw other than document files in this PR. Reverted.

**Review by RatkoVasic-NOAA — APPROVED** [review]
I followed this convention on tier1 machines. Looks good.

**Comment by climbfuji** [comment]
> > The jet updates have nothing to do with this PR and should be removed ...
> > @climbfuji , sorry, I saw other than document files in this PR. Reverted.

Yeah, apologies. This PR originally had `develop` as the target/base branch and showed all the changes between (release/1.8.0 plus this PR) and develop. Steve pointed that out and I changed the base branch to release/1.8.0. Let's do all the NOAA site config and doc updates in a separate PR from you, ok?

**Pull request merged**